### PR TITLE
Propse var reorganisation

### PR DESCRIPTION
**Pull request by gwarf**
Move vars from `vars/main.yml` to `defaults/main.yml` so that all the variables can easily be overridden when including the role in another playbook.
Related to #14

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Requirements
 Role Variables
 --------------
 
-Variables are used to tune the Rails application that runs Discourse. These are in a template, and set in the `defaults/main.yml` and `vars/main.yml` file. Please read them and set them accordingly in `vars/main.yml`
+Variables are used to tune the Rails application that runs Discourse. These are in a template, and set in the `defaults/main.yml` file. Please read them and set them accordingly in your playbook
```
Comment (brucellino) [inline README.md]: Following our discussion in #14 I suggest we keep the original text, but remove `vars/main.yml` from change control.

File: defaults/main.yml [hunk 2]
```diff
@@ -10,15 +10,27 @@ prerequisites:
     - git
     - vim
 
-# Buffer size in MB
-db_shared_buffer_size: 256
-unicorn_workers: 2
+# discourse_hostname:
```
Comment (brucellino) [inline defaults/main.yml:13]: These are sane defaults, so I think they can stay.

File: defaults/main.yml [hunk 3]
```diff
@@ -10,15 +10,27 @@ prerequisites:
...
+# These are the varibles needed by the discourse container
+# Bufffer size in MB
+db_shared_buffer_size: 512
+unicorn_workers: 4
+
 # email settings
-# change this - it makes no sense
-developer_emails: nobody@nowhere.com
-smtp_address: smtp.gmail.com
+developer_emails: 'brucellino@gmail.com'
```
Comment (brucellino) [inline defaults/main.yml]: This is not a sane default though

File: defaults/main.yml [hunk 4]
```diff
@@ -10,15 +10,27 @@ prerequisites:
...
+smtp_address: smtp.mailgun.org
 smtp_port: 587
-smtp_username: admin@example.com
+smtp_username: postmaster@sandbox6c9929c31d284bdba0a154af210f7130.mailgun.org
```
Comment (brucellino) [inline defaults/main.yml]: neither is this.

File: defaults/main.yml [hunk 5]
```diff
@@ -28,5 +40,26 @@ discourse_templates:
   - sshd
   - web.ratelimited
 
+# add plugins here.
+# only the url is used in the template app.yml.j2.
+# The other fields are to help you remember what they are and why you
+# what them.
+discourse_plugins:
```
Comment (brucellino) [inline defaults/main.yml:47]: It doesn't make sense to have a long list of default plugins, every discourse is special :snowflake: but perhaps these are good examples of defaults. I'm ok with the list.

**Review by brucellino — CHANGES_REQUESTED** [review]

**Comment by gwarf** [comment]
So as I was unsure I proposed to move all the contents from `vars/main.yml` to `defaults/main.yml`, but once the plan will be OK I will clean and use only the sane default values you mentioned. I was just confused by this embedding of non-easily overridable deployment-specific conf in the role.

**Review by brucellino — APPROVED** [review]
With a minor tweak of the README, I am staisfied

**Comment by gwarf** [comment]
I've updated some of the defaults, let me know if it's OK with you and please squash/merge.
This should also close #14.

**Review by brucellino — APPROVED** [review]
The defaults make sense, so I'm ok with having the vars there.

**Pull request merged**